### ISSUE TITLE
Crash in SLPvectorizer on assertion  "trying to erase instruction with users."

### ISSUE THREAD

**Issue by valerydmit** [llvm/llvm-project#79229]
to reproduce:
opt -passes=slp-vectorizer -mcpu=x86-64-v3 -disable-output f.ll
[f.txt](https://github.com/llvm/llvm-project/files/14030909/f.txt)

f.ll attached as f.txt.

Fails after 
```
commit ac254fc055980219b30821c3717c6b7db0fbbc46
Author: Alexey Bataev <a.bataev@outlook.com>
Date:   Tue Apr 18 04:16:58 2023 -0700

    [SLP]Improve tryToGatherExtractElements by using per-register analysis.
    
    Currently tryToGatherExtractElements function analyzes the whole vector,
    regrdless number of actual registers, used in this vector. It may
    prevent some optimizations, because per-register analysis may allow to
    simplify the final code by reusing more already emitted vectors and
    better shuffles.
    
    Differential Revision: https://reviews.llvm.org/D148855
```

**Comment by alexey-bataev** [comment]
Fixed in 48bbd7658710ef1699bf2a6532ff5830230aacc5

**Comment by valerydmit** [comment]
[crash.txt](https://github.com/llvm/llvm-project/files/14044996/crash.txt)
Another test case. Fails on the same assertion.

**Comment by alexey-bataev** [comment]
Fixed in 6fe21bc1dac883efa0dfa807f327048ae9969b81

**Comment by valerydmit** [comment]
Thank you for addressing it promptly, @alexey-bataev .
